controllers/vspherecluster_reconciler.go

```diff
@@ -117,18 +118,90 @@ func (r *clusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return r.reconcileDelete(ctx, clusterContext)
 	}
 
```

Comment:
Why did we remove this?

I think it's part of the contract: https://cluster-api.sigs.k8s.io/developer/providers/contracts/infra-cluster#typical-infracluster-reconciliation-workflow

Reply:
Oh, very good finding.

I did it to ensure the first condition at least always get set so we can rely on it when calculating Ready.

But then it probably is better to revert this and instead adjust the patch logic to be okay with none of the conditions ready depends on are set.

Reply:
Yup. Or return before the defer. How does this work in the supervisor cluster reconciler?